katportalclient/test/test_client.py

```diff
@@ -779,147 +984,296 @@ def test_future_targets(self):
+            start_time='2017-02-07 08:47:22',
+            end_time='2017-02-07 08:47:22')
+        self.assertEquals(
+            userlog,
+            {u'other_metadata': u'[]', u'user_id': u'1', u'attachments': u'[]',
+             u'tags': u'[]', u'timestamp': u'2017-02-07 08:47:22',
+             u'start_time': u'2017-02-07 00:00:00', u'modified': u'',
+             u'content': u'test content',
+             u'parent_id': u'',
+             u'user': {u'email': u'cam@ska.ac.za', u'name': u'CAM', u'id': 1},
+             u'attachment_count': u'0', u'id': u'40',
+             u'end_time': u'2017-02-07 23:59:59'})
 
-        result = yield self._portal_client._get_target_descriptions(
-            'Moon,Sun', 1, 2, 3, 0, 'test config label')
+    @gen_test
+    def test_modify_userlog(self):
+        """Test userlog creation"""
+        # fake a login
+        self._portal_client._session_id = 'some token'
+        self._portal_client._current_user_id = 1
+        # then list userlogs
+
+        userlogs_base_url = self._portal_client.sitemap['userlogs']
+        authorized_fetch_future = gen.Future()
+        self._portal_client.authorized_fetch = mock.MagicMock(
+            return_value=authorized_fetch_future)
+        auth_fetch_result = HTTPResponse(
+            HTTPRequest(userlogs_base_url), 200,
+            buffer=StringIO.StringIO(
+                r"""
+                {
+                    "other_metadata": "[]",
+                    "user_id": "1",
+                    "attachments": "[]",
+                    "tags": "[]",
+                    "timestamp": "2017-02-07 08:47:22",
+                    "start_time": "2017-02-07 00:00:00",
+                    "modified": "",
+                    "content": "test content modified",
+                    "parent_id": "",
+                    "user": {"email": "cam@ska.ac.za", "id": 1, "name": "CAM"},
+                    "attachment_count": "0",
+                    "id": "40",
+                    "end_time": "2017-02-07 23:59:59"
+                }
+                """))
+        authorized_fetch_future.set_result(auth_fetch_result)
+
+        userlog_to_modify = {
+            'other_metadata': [],
+            'user_id': 1,
+            'attachments': [],
+            'tags': '[]',
+            'timestamp': '2017-02-07 08:47:22',
+            'start_time': '2017-02-07 00:00:00',
+            'modified': '',
+            'content': 'katportalclient userlog modified content!',
+            'parent_id': '',
+            'user': {'email': 'cam@ska.ac.za', 'id': 1, 'name': 'CAM'},
+            'attachment_count': 0,
+            'id': 40,
+            'end_time': '2017-02-07 23:59:59'
+        }
+        userlog = yield self._portal_client.modify_userlog(userlog_to_modify, [1,2,3])
         self.assertEquals(
-            result,
-            [{u'name': u'Moon', u'testvar': u'test variable 1'},
-             {u'name': u'Sun', u'testvar': u'test variable 2'}])
+            userlog,
```

Comment:
This doesn't really test that the expected modification happened, just that there was a call to authorized_fetch, which is returning your own mock content.  E.g. the [1, 2, 3] tag_ids don't show up in the actual output.  Suggest you also check the `body` param passed to that mock call is the modified version of the user log.